haystack/components/generators/hugging_face_api.py

```diff
@@ -217,18 +218,31 @@ def _stream_and_build_response(
         self, hf_output: Iterable["TextGenerationStreamOutput"], streaming_callback: Callable[[StreamingChunk], None]
     ):
         chunks: List[StreamingChunk] = []
+        first_chunk_time = None
+
         for chunk in hf_output:
             token: TextGenerationOutputToken = chunk.token
             if token.special:
                 continue
-            chunk_metadata = {**asdict(token), **(asdict(chunk.details) if chunk.details else {})}
+
+            chunk_metadata = {
+                **asdict(token),
+                **(asdict(chunk.details) if chunk.details else {}),
+                "received_at": datetime.now().isoformat(),
+            }
+
+            if first_chunk_time is None:
+                first_chunk_time = chunk_metadata["received_at"]
+
```

Comment:
same as above - could we simplify it?

Reply:
Great suggestions, have another look. Let's do the same for other generators in integrations.